src/main/java/de/tum/cit/aet/artemis/exercise/service/ExerciseService.java

```diff
@@ -809,4 +821,52 @@ public Map<ExerciseType, Long> countByCourseIdGroupByType(Long courseId) {
 
         return Arrays.stream(ExerciseType.values()).collect(Collectors.toMap(type -> type, type -> exerciseTypeCountMap.getOrDefault(type, 0L)));
     }
+
+    /**
+     * Checks if the due date of an exercise has changed and updates related slides if needed.
+     * This method should be called after saving an updated exercise.
+     *
+     * @param originalExercise The original exercise before the update
+     * @param updatedExercise  The updated exercise after the update
+     */
+    public void handleDueDateChange(Exercise originalExercise, Exercise updatedExercise) {
+        ZonedDateTime originalDueDate = originalExercise.getDueDate();
+        ZonedDateTime updatedDueDate = updatedExercise.getDueDate();
+
+        // Check if the due date has changed
+        if (updatedDueDate != null && (originalDueDate == null || !originalDueDate.equals(updatedDueDate))) {
+
+            updateRelatedSlides(updatedExercise);
+        }
+
+    }
+
+    /**
+     * Updates the hidden date of slides associated with the given exercise to match the exercise's due date.
+     * This method should only be called when an exercise's due date has changed.
+     *
+     * @param exercise The exercise whose due date has changed
+     */
+    public void updateRelatedSlides(Exercise exercise) {
+        if (exercise.getDueDate() == null) {
+            return;
+        }
+
+        List<Slide> relatedSlides = slideRepository.findByExerciseId(exercise.getId());
+        if (relatedSlides.isEmpty()) {
+            return;
+        }
+
+        log.debug("Updating hidden date for {} slides related to exercise {}", relatedSlides.size(), exercise.getId());
+
+        Date newHiddenDate = Date.from(exercise.getDueDate().toInstant());
+
+        for (Slide slide : relatedSlides) {
+            slide.setHidden(newHiddenDate);
+            slideRepository.save(slide);
+
+            // Schedule the slide to be unhidden at the new date
+            slideUnhideService.handleSlideHiddenUpdate(slide);
+        }
```

Comment:
While this suggestion iterates over the same array twice instead of once, the call to update the slides is batched, hence faster.

```suggestion
        relatedSlides.forEach(slide -> slide.setHidden(newHiddenDate));
        slideRepository.saveAll(relatedSlides);

        // Schedule the slide to be unhidden at the new date
        relatedSlides.forEach(slideUnhideService::handleSlideHiddenUpdate);
```

Reply:
Thank you very much for the recommendation, I will apply this.